### ISSUE TITLE
Update Readme and Improve Documentation

### ISSUE THREAD

**Issue by DarkSideOfTheMat** [bluebonnet-data/bbd#13]
- [ ] Update the Readme.md to further explain library contents (currently missing geocoder module and some explanation for gis module). 

- [ ] The library needs some kind of documentation to add to the project for full description of project contents (readthedocs.io/github wiki).

**Comment by boom-roasted** [comment]
As far as the README goes, it seems like we have two options:
* One big master readme on the front page that explains everything (each module has it's own section?)
* One readme per module

Personally I think that having one big master readme is better because, while it may be a little more unsightly, it is a lot easier to search though (i.e. ctrl+F) and find the topics you are looking for.

What do you think?

**Comment by DarkSideOfTheMat** [comment]
Yeah I think the idea of 1 bigger readme is probably the way to go. I added a little table of contents to the walkthrough section when I added the geocoder section in #14
